File: ios/Classes/Modules/InstabugLogApi.m
#import "Instabug.h"
#import "InstabugLogApi.h"

@implementation InstabugLogApi

- (void)logVerboseMessage:(NSString *)message error:(FlutterError *_Nullable *_Nonnull)error {
    [IBGLog logVerbose:message];
}

- (void)logDebugMessage:(NSString *)message error:(FlutterError *_Nullable *_Nonnull)error {
    [IBGLog logDebug:message];
}

- (void)logInfoMessage:(NSString *)message error:(FlutterError *_Nullable *_Nonnull)error {
    [IBGLog logInfo:message];
}

- (void)logWarnMessage:(NSString *)message error:(FlutterError *_Nullable *_Nonnull)error {
    [IBGLog logWarn:message];
}

- (void)logErrorMessage:(NSString *)message error:(FlutterError *_Nullable *_Nonnull)error {
    [IBGLog logError:message];
}

- (void)clearAllLogsWithError:(FlutterError * _Nullable __autoreleasing * _Nonnull)error {
    [IBGLog clearAllLogs];
}

@end

File: android/src/main/java/com/instabug/flutter/modules/InstabugLogApi.java
package com.instabug.flutter.modules;

import androidx.annotation.NonNull;

import com.instabug.flutter.generated.InstabugLogPigeon;
import com.instabug.library.logging.InstabugLog;

import io.flutter.plugin.common.BinaryMessenger;

public class InstabugLogApi implements InstabugLogPigeon.InstabugLogHostApi {
    public static void init(BinaryMessenger messenger) {
        final InstabugLogApi api = new InstabugLogApi();
        InstabugLogPigeon.InstabugLogHostApi.setup(messenger, api);
    }

    @Override
    public void logVerbose(@NonNull String message) {
        InstabugLog.v(message);
    }

    @Override
    public void logDebug(@NonNull String message) {
        InstabugLog.d(message);
    }

    @Override
    public void logInfo(@NonNull String message) {
        InstabugLog.i(message);
    }

    @Override
    public void logWarn(@NonNull String message) {
        InstabugLog.w(message);
    }

    @Override
    public void logError(@NonNull String message) {
        InstabugLog.e(message);
    }

    @Override
    public void clearAllLogs() {
        InstabugLog.clearLogs();
    }
}
